### PR TITLE
Create project structure

### PR DESCRIPTION
**Pull request by kfurgol**
**Description**

Changes proposed in this pull request:

- Created 4 catalogs for new components:
  - gateway
  - director
  - registry
  - healthchecker

- Each catalog consists of:
  - .gitignore file
  - before-commit.sh script
  - Dockerfile (base image alpine:3.9)
  - Makefile
  - Gopkg.toml and Gopkg.lock (dep inited with version 0.5.2)
  - cmd directory with main.go file
  - README.md file

- Updated CODEOWNERS file

**Related issue(s)**
#2

### CODE REVIEW

**Comment by PK85** [comment]
we decided to have director and not app-director as that component will contains runtimes and applications

**Comment by kfurgol** [comment]
> we decided to have director and not app-director as that component will contains runtimes and applications

@PK85 
I've renamed director along with other components, as we discussed

File: components/director/before-commit.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+if [ $(echo ${#goImportsResult}) != 0 ]; then
+	echo -e "${RED}✗ goimports and fmt${NC}\n$goImportsResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ goimports and fmt${NC}"
+fi
+
+##
+# GO VET
+##
```
Comment (pkosiec) [inline components/director/before-commit.sh]: I would use the `before-commit.sh` version with `errcheck` - please see the script for 
 e.g. Console Backend Service

Comment (kfurgol) [reply components/director/before-commit.sh]: done

**Pull request merged**